bin/holdings_deleter.rb

```diff
@@ -0,0 +1,132 @@
+# frozen_string_literal: true
+
+require "services"
+require "cluster"
+require "date"
+
+Services.mongo!
+
+STRING_FIELDS = [
+  "organization",
+  "local_id",
+  "enum_chron",
+  "n_enum",
+  "n_chron",
+  "status",
+  "condition",
+  "mono_multi_serial",
+  "country_code",
+  "uuid",
+  "issn"
+].freeze
+
+DATE_FIELDS  = ["date_received"].freeze
+INT_FIELDS   = ["ocn"].freeze
+FLOAT_FIELDS = ["weight"].freeze
+BOOL_FIELDS  = ["gov_doc_flag"].freeze
+CTRL_FIELDS  = ["noop", "verbose", "leave_empties"].freeze
+
+# Takes simple pairs of key=>value and constructs a update_many...pull query
+# based on those criteria.
+# Example:
+# bundle exec ruby bin/delete_matching_holdings.rb organization foo date_received 2020-01-01
+class HoldingsDeleter
+  attr_reader :matching_criteria
+
+  def initialize(args)
+    @matching_criteria = {}
+
+    Hash[*args].each do |key, val|
+      val_r = recast_val(key, val)
+      unless val_r.nil?
+        @matching_criteria[key] = val_r
+      end
+    end
+
+    if @matching_criteria.empty?
+      raise "No matching criteria given."
+    end
+
+    putsif "Criteria:"
+    @matching_criteria.each do |k, v|
+      putsif "#{k}:\t#{v} (#{v.class})"
+    end
+  end
+
+  def run
+    if @noop
+      puts "noop!"
+      return nil
+    end
+
+    result = Cluster.collection.update_many(
+      {},
+      { "$pull": { "holdings": { "$and": [@matching_criteria] } } }
```

Comment:
$pull should definitely be faster. Another option would be 
`Cluster.where( <criteria>).no_timeout.each do |cluster|
    ...delete the necessary holdings...
    ...delete the cluster if it's empty...
`